src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -420,6 +420,13 @@ private async Task<bool> AreCentralVersionRequirementsSatisfiedAsync()
 
                 return false;
             }
+            IEnumerable<LibraryDependency> packageReferencedDependenciesWithoutCentralVersionDefined = _request.Project.TargetFrameworks.SelectMany(tfm => tfm.Dependencies.Where(d=>d.LibraryRange.VersionRange == null));
```

Comment:
Could probably use Any check on this whole thing without assigning a variable.

There's some room for formatting improvements too.

Reply:
Fixed formatting. The variable is used to get the package Ids that need to be fixed.